activerecord/lib/active_record/normalization.rb

```diff
@@ -23,12 +25,12 @@ module Normalization
     #   legacy_user.normalize_attribute(:email)
     #   legacy_user.email # => "cruise-control@example.com"
     #   legacy_user.save
-    def normalize_attribute(name)
-      # Treat the value as a new, unnormalized value.
-      self[name] = self[name]
-    end
 
     module ClassMethods
+      ##
+      # :method: normalizes
+      # :call-seq: normalizes(*names, with:, apply_to_nil: false)
+      #
       # Declares a normalization for one or more attributes. The normalization
       # is applied when the attribute is assigned or updated, and the normalized
       # value will be persisted to the database. The normalization is also
```

Comment:
This method doc includes some caveats and examples specific to Active Record, but perhaps we can separate those out.  Something like:

```ruby
# See ActiveModel::Normalization::ClassMethods#normalizes.
#
# To prevent confusion, normalization will not be applied
# when the attribute is fetched from the database. This means that if a
# record was persisted before the normalization was declared, the record's
# attribute will not be normalized until either it is assigned a new
# value, or it is explicitly migrated via Normalization#normalize_attribute.
#
# Be aware that if your app was created before Rails 7.1, and your app
# marshals instances of the targeted model (for example, when caching),
# then you should set ActiveRecord.marshalling_format_version to +7.1+ or
# higher via either <tt>config.load_defaults 7.1</tt> or
# <tt>config.active_record.marshalling_format_version = 7.1</tt>.
# Otherwise, +Marshal+ may attempt to serialize the normalization +Proc+
# and raise +TypeError+.
#
# ==== Examples
#
# ...
```